File: 240423/이상한 폭탄 2/strange-bomb-2.py
N, K = map(int, input().split())
nums = [int(input()) for _ in range(N)]

answer = -1
explode_nums = []
nums_cnt = dict.fromkeys(sorted(set(nums)), 0)

for num_idx, num in enumerate(nums):
    if num_idx < K and num in nums[num_idx + 1:num_idx + K + 1]:
        explode_nums.append(num)
    elif num_idx + K <= len(nums) - 1 and num in nums[num_idx + 1:]:
        explode_nums.append(num)
    elif K <= num_idx < len(nums) + K - 1 and num in nums[num_idx - K // 2:num_idx + K // 2]:
        explode_nums.append(num)

print(max(set(explode_nums)))

File: 260310/이상한 폭탄 2/strange-bomb-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, K;
    cin >> N >> K;

    vector<int> nums(N);
    for (int i = 0; i < N; i++) 
    {
        cin >> nums[i];
    }

    int answer = -1;

    for (int i = 0; i < N; i++) 
    {
        int bomb_num = -1;

        for (int j = i + 1; j <= i + K; j++) 
        {
            if (j > N - 1) continue;

            if (nums[i] == nums[j]) 
            {
                bomb_num = nums[i];
            }
        }

        answer = max(answer, bomb_num);
    }

    cout << answer << '\n';
    return 0;
}
